nrf-hal-common/src/saadc.rs

```diff
@@ -65,18 +84,29 @@ impl Saadc {
     }
 }
 
+// TODO problems:
```

Comment:
This is generally okay, IMO. if you'd like, you can add asserts in the doc comments that these values are correct, like this (note, I used `'''` instead of backticks so it renders right in GH, you should use backticks in real doc comments):

```rust
/// This is an SaadcConfig
///
/// Example:
///
/// '''rust
/// use hal::saadc::{SaadcConfig, Resolution, Reference};
/// let foo = SaadcConfig::default();
///
/// assert_eq!(foo.resolution, Resolution::OVER8X);
///
/// // this assert is hidden from the rendered docs
/// # assert_eq!(foo.reference, Reference::VDD1_4);
/// '''
```